packages/api/src/hooks/p2p/useOrderList.ts

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import useInfiniteQuery from '../../useInfiniteQuery';
+import useAuthorize from '../useAuthorize';
+
+/** This custom hook returns a list of orders under the current client. */
+const useOrderList = (
+    payload?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_order_list'>>[1]>['payload'],
+    config?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_order_list'>>[1]>['options']
+) => {
+    const { isSuccess } = useAuthorize();
+    const { data, fetchNextPage, ...rest } = useInfiniteQuery('p2p_order_list', {
+        payload: { ...payload, offset: payload?.offset, limit: payload?.limit },
+        options: {
+            getNextPageParam: (lastPage, pages) => {
+                if (!lastPage?.p2p_order_list?.list) return;
+
+                return pages.length;
+            },
+            enabled: isSuccess && (config?.enabled === undefined || config.enabled),
+        },
+    });
+
+    // Flatten the data array
+    const flatten_data = React.useMemo(() => {
+        if (!data?.pages?.length) return;
+
+        return data?.pages?.flatMap(page => page?.p2p_order_list?.list);
+    }, [data?.pages]);
+
+    // Additional p2p_order_list data
+    const modified_data = React.useMemo(() => {
+        if (!flatten_data?.length) return undefined;
+
+        return flatten_data.map(advert => ({
+            ...advert,
+            /** Details of the advert for this order. */
+            advert_details: {
+                ...advert?.advert_details,
+                /** Indicates if this is block trade advert or not. */
+                block_trade: Boolean(advert?.advert_details?.block_trade),
+            },
+            /** Details of the advertiser for this order. */
+            advertiser_details: {
+                ...advert?.advertiser_details,
+                /** Indicates if the advertiser is currently online. */
+                is_online: Boolean(advert?.advertiser_details?.is_online),
+                /** Indicates that the advertiser was recommended in the most recent review by the current user. */
+                is_recommended: Boolean(advert?.advertiser_details?.is_recommended),
+            },
+            /** Details of the client who created the order. */
+            client_details: {
+                ...advert?.client_details,
+                /** Indicates if the advertiser is currently online. */
+                is_online: Boolean(advert?.advertiser_details?.is_online),
+                /** Indicates that the advertiser was recommended in the most recent review by the current user. */
+                is_recommended: Boolean(advert?.advertiser_details?.is_recommended),
+            },
+            /** The epoch time of the order completion. */
+            completion_time: advert?.completion_time ? new Date(advert.completion_time) : undefined,
```

Comment:
i tot no conversion to `Date` 😱

Reply:
oh i thought i deleted this, will add it 👍